### ISSUE TITLE
Need help understanding this example

### ISSUE THREAD

**Issue by tandrewnichols** [JanPeter/neo4js#8]
More of a question than an issue . . .

I'm not using flow, so I'm looking at the without-flow example, but I won't have access to decorators, as node doesn't support them (and I don't want to transpile server side code with babel). How do I call `@model(User)` without using decorators? I tried looking at the generated code, but babel transpiling is always hard to follow.

I've done just a cursory search on Google, and I'll keep looking, but if you happen to know offhand what that looks like in es2015 code, let me know.

**Comment by JanPeter** [comment]
I'm going to write an example today 👍

**Comment by JanPeter** [comment]
This actually needed a little more than just writing an example. I built a workaround into neo4js, have a look at https://neo4.js.org/docs/vanilla-node-guide.html on how to use neo4js without decorators. You'll need to update your neo4js dependency. I'm not a fan of it, but it works, let me know what you think about it and feel free to reopen this issue if you have any troubles with it!

**Comment by tandrewnichols** [comment]
Yeah, I definitely see how that looks nicer with decorators. The `model` one seems the best because it almost reads to me as `connect(User, UserInstance)`. The other ones are harder to reason about, but not completely unusable.

**Comment by JanPeter** [comment]
Maybe I'll rename this things! I already thought about connect. I think I could also give the relation "connect" the same name instead of two different (src, dest).

**Comment by tandrewnichols** [comment]
Yeah, maybe something like `connect(User, '[tasks]').to(Task).via('created')` (where the brackets around "tasks" indicate the "many" part . . . other ways of expressing many are possible too, this was just the first that occurred to me).

**Comment by tandrewnichols** [comment]
If you can't tell, I'm a fan of the "fluent" chaining syntax . . . :)

**Comment by JanPeter** [comment]
This (`connect(User, '[tasks]').to(Task).via('created')`) won't work because I need to know on which property of the ModelInstance I need to add the relation methods. Thus I went with decorators to connect the ModelInstance class properties to the desired relation like this:
 
```
const UserProjectRelation = relation("assigned")
  .src.hasMany(Project)
  .dest.hasMany(User);

@model(User)
class UserInstance extends ModelInstance<UserProps> {
  @src(UserProjectRelation)
  projects: HasManyActions<ProjectProps, ProjectInstance>;
}
```

But when I look at this, I only need to add the class properties directly to the declaration of the relation itself like this:

```
relation(Project, "assignedUsers")
  .via("assigned")
  .hasMany(User, "projects");
```

Which will create the relation methods on `<ProjectInstance>.assignedUsers` and `<UserInstance>.projects`. With that I would only need to use `connect(Model, ModelInstance)` and we are all set 👍 . The order wether you write `via` or `hasMany/hasOne` after `relation()` shouldn't matter. As it also doesn't matter with the current implementation of `.src` and `.dest`.

I think the keyword `connect` would also make sense in this case like the following:

```
connect(Project, "assignedUsers")
  .via("assigned")
  .hasMany(User, "projects");
```

This is actually less code and better readable, at least in my mind :P
In fact the only reason I didn't do that in the first place is, that I wouldn't get any type hints from flow. So I went with decorators, which are slightly more complicated to reason about the relation.

I think I'll change the API for the non-decorator and non-flow approach to that. What do you think about that? Also we should either change the title of this issue or move this discussion to a new issue.

**Comment by tandrewnichols** [comment]
Yeah, I like that pretty well. It does seem like a different issue though, especially since you've already added a non-decorator example, which satisfies this issue.

**Comment by JanPeter** [comment]
Well, let's move it to a new issue then :)
